src/Plugin.php

```diff
@@ -24,7 +24,7 @@
      * @param callable         $next    Next middleware in the chain, the request is passed as the first argument
      * @param callable         $first   First middleware in the chain, used to to restart a request
      *
-     * @return Promise
+     * @return Promise Resolves a PSR-7 Response or fails with an Http\Client\Exception (The same as HttpAsyncClient).
```

Comment:
Is this a BC break ? (For me no, it's more a clarification but i dont know if this interface is already used but other packages than one we have under control ?)